swift-recon.py

```diff
@@ -351,9 +348,18 @@ def main():
             maas_common.status_err('no ring provided to check')
         stats = swift_replication(args.ring)
     else:
-        maas_common.status_err(
-            'unrecognized command "{0}"'.format(args.recon)
-            )
+        raise TypeError('unrecognized command "{0}"'.format(args.recon))
+    return stats
+
+
+def main():
+    parser = make_parser()
+    args = parser.parse_args()
+
+    try:
+        stats = get_stats_from(args)
+    except (ValueError, TypeError) as e:
```

Comment:
Thinking about it again, I wonder if this is a good idea. Catching built-in exceptions makes it more likely that something unintended might be caught, in the sense that str(e) might not be very specific. Maybe it would be better use status_err instead of raising exceptions.